File: AES.rb
require 'openssl'
require 'base64'

def aes_encrypt(input, pass)
  aes = OpenSSL::Cipher.new('AES-128-ECB')
  hash = OpenSSL::Digest::MD5.digest(pass)

  aes.encrypt
  aes.key = hash
  encrypted = aes.update(input) + aes.final
  Base64.encode64(encrypted)
rescue => e
  "ERROR encrypt: #{e.message}"
end

def aes_decrypt(input, pass)
  aes = OpenSSL::Cipher.new('AES-128-ECB')
  hash = OpenSSL::Digest::MD5.digest(pass)

  aes.decrypt
  aes.key = hash
  decrypted = aes.update(Base64.decode64(input)) + aes.final
  decrypted
rescue => e
  "ERROR decrypt: #{e.message}"
end


puts 'Enter text to encrypt:'
text = gets.chomp
puts 'Enter password:'
password = gets.chomp

if text.empty? || password.empty?
  puts 'Text or password cannot be empty.'
else
  encrypted_text = aes_encrypt(text, password)
  puts "Encrypted text: #{encrypted_text}"

  decrypted_text = aes_decrypt(encrypted_text, password)
  puts "Decrypted text: #{decrypted_text}"
end

File: AES.py
from Crypto.Cipher import AES
from Crypto.Hash import MD5
import base64

def create_key(password):
    hash_obj = MD5.new()
    hash_obj.update(password.encode('utf-8'))
    return hash_obj.digest()

def aes_encrypt(input_text, password):
    try:
        key = create_key(password)
        cipher = AES.new(key, AES.MODE_ECB)
        padded_input = input_text + (AES.block_size - len(input_text) % AES.block_size) * chr(AES.block_size - len(input_text) % AES.block_size)
        encrypted = cipher.encrypt(padded_input.encode('utf-8'))
        return base64.b64encode(encrypted).decode('utf-8')
    except Exception as e:
        return f"ERROR encrypt: {e}"

def aes_decrypt(input_text, password):
    try:
        key = create_key(password)
        cipher = AES.new(key, AES.MODE_ECB)
        decrypted = cipher.decrypt(base64.b64decode(input_text))
        padding_length = decrypted[-1]
        return decrypted[:-padding_length].decode('utf-8')
    except Exception as e:
        return f"ERROR decrypt: {e}"

if __name__ == "__main__":
    text = input("Enter text to encrypt: ")
    password = input("Enter password: ")

    if not text or not password:
        print("Text or password cannot be empty.")
    else:
        encrypted_text = aes_encrypt(text, password)
        print(f"Encrypted text: {encrypted_text}")
        decrypted_text = aes_decrypt(encrypted_text, password)
        print(f"Decrypted text: {decrypted_text}")
